model/task/db.go

```diff
@@ -812,6 +812,63 @@ func GetRecentTaskStats(period time.Duration, nameKey string) ([]StatusItem, err
 	return result, nil
 }
 
+// ByExecutionTasksAndMaxExecution returns the tasks corresponding to the passed in taskIds and execution,
+// or the most recent executions of those tasks if they do not have a matching execution
+func ByExecutionTasksAndMaxExecution(taskIds []*string, execution int) ([]Task, error) {
+	pipeline := []bson.M{}
+	match := bson.M{
+		"$match": bson.M{
+			IdKey: bson.M{
+				"$in": taskIds,
+			},
+			ExecutionKey: bson.M{
+				"$lte": execution,
+			},
+		},
+	}
+	pipeline = append(pipeline, match)
+	// Get the taskIds that were not found in the previous match stage
+	result := []Task{}
+	if err := Aggregate(pipeline, &result); err != nil {
+		return nil, errors.Wrap(err, "Error finding tasks in task collection")
+	}
+	taskMap := map[string]bool{}
+	for _, t := range result {
+		taskMap[t.Id] = true
+	}
+	missingTasks := []string{}
+	for _, taskId := range taskIds {
+		tId := utility.FromStringPtr(taskId)
+		if _, ok := taskMap[tId]; !ok {
+			missingTasks = append(missingTasks, tId)
+		}
+	}
+	if len(missingTasks) > 0 {
+		oldTasks := []Task{}
+		oldTaskPipeline := []bson.M{}
+		match = bson.M{
+			"$match": bson.M{
+				OldTaskIdKey: bson.M{
+					"$in": missingTasks,
+				},
+				ExecutionKey: bson.M{
+					"$lte": execution,
+				},
+			},
+		}
+		oldTaskPipeline = append(oldTaskPipeline, match)
+		if err := db.Aggregate(OldCollection, oldTaskPipeline, &oldTasks); err != nil {
+			return nil, errors.Wrap(err, "error finding tasks in old tasks collection")
+		}
+
+		result = append(result, oldTasks...)
+	}
+	if len(result) == 0 {
+		return nil, nil
+	}
+	return result, nil
```

Comment:
Should we sort this on anything?  / does it not matter that we aren't doing WithDisplayStatus anymore 🤔

Reply:
I don't think so we don't sort execution tasks anywhere else. The only place these results need to be sorted is on the execution tasks tab but that is handled client side.